test/externalTests.sh

```diff
@@ -44,45 +44,42 @@ function test_truffle
     echo "Running $name tests..."
     DIR=$(mktemp -d)
     (
+      cd "$DIR"
+      git clone --depth 1 -b v0.5.0 https://github.com/ethereum/solc-js.git solc
+      SOLCVERSION="UNDEFINED"
+
+      cd solc
+      npm install
+      cp "$SOLJSON" soljson.js
+      SOLCVERSION=$(./solcjs --version)
+      cd ..
+      echo "Using solcjs version $SOLCVERSION"
+
       if [ -n "$branch" ]
       then
         echo "Cloning $branch of $repo..."
-        git clone --depth 1 "$repo" -b "$branch" "$DIR"
+        git clone --depth 1 "$repo" -b "$branch" "$DIR/ext"
       else
         echo "Cloning $repo..."
-        git clone --depth 1 "$repo" "$DIR"
+        git clone --depth 1 "$repo" "$DIR/ext"
       fi
-      cd "$DIR"
+      cd ext
       echo "Current commit hash: `git rev-parse HEAD`"
       npm install
-      # Replace solc package by v0.5.0
-      for d in node_modules node_modules/truffle/node_modules
-      do
-      (
-        if [ -d "$d" ]
-        then
-          cd $d
-          rm -rf solc
-          git clone --depth 1 -b v0.5.0 https://github.com/ethereum/solc-js.git solc
-          cp "$SOLJSON" solc/
-        fi
-      )
-      done
       if [ "$name" == "Zeppelin" -o "$name" == "Gnosis" ]; then
         echo "Replaced fixed-version pragmas..."
         # Replace fixed-version pragmas in Gnosis (part of Consensys best practice)
         find contracts test -name '*.sol' -type f -print0 | xargs -0 sed -i -e 's/pragma solidity [\^0-9\.]*/pragma solidity >=0.0/'
       fi
-      assertsol="node_modules/truffle/build/Assert.sol"
-      if [ -f "$assertsol" ]
-      then
-        echo "Replace Truffle's Assert.sol with a known good version"
-        rm "$assertsol"
-        wget https://raw.githubusercontent.com/trufflesuite/truffle-core/ef31bcaa15dbd9bd0f6a0070a5c63f271cde2dbc/lib/testing/Assert.sol -o "$assertsol"
-      fi
       # Change "compileStandard" to "compile"
       sed -i s/solc.compileStandard/solc.compile/ "node_modules/truffle/build/cli.bundled.js"
+      cat >> truffle*.js <<EOF
```

Comment:
Also please add a comment above this line stating we require Truffle 5.x.

There's no need to add a runtime check to verify it is Truffle 5.x until we run into projects which don't use it.